### PR TITLE
plugins/typst-vim: init plugin + test

### PR DESCRIPTION
**Pull request by GaetanLepage**
Add support for the [`typst.vim`](https://github.com/kaarmu/typst.vim) plugin.

### CODE REVIEW

**Review by traxys — CHANGES_REQUESTED** [review]
We should either set the `typst_cmd` to a nix store path, or add `typst` to the extraPackages I think. The first one may be more flexible as you can easily customize the typst version you have (or set it to null to manage it externally)

File: plugins/languages/typst/typst-vim.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  config = mkIf cfg.enable {
+    extraPlugins = [cfg.package];
+
+    # Add the typst compiler to nixvim packages
+    extraPackages = with pkgs; [typst];
+
+    maps.normal = with cfg.keymaps;
+      helpers.mkModeMaps {inherit silent;} {
+        ${watch} = ":TypstWatch<CR>";
```
Comment (traxys) [inline plugins/languages/typst/typst-vim.nix:39]: Does this work when watch `watch` is null ? I'm not sure

Comment (GaetanLepage) [reply plugins/languages/typst/typst-vim.nix:39]: Yes it works. Cf the `empty` test case.
I also tested locally without any mapping.
However, maybe it is overkill to wrap keybindings for this plugin. It does not really add any value...

Comment (traxys) [reply plugins/languages/typst/typst-vim.nix:39]: Ah sorry I though the keymaps were optional. It's not really a problem to provide keymaps, it helps discovery of options.

**Review by traxys — APPROVED** [review]

**Pull request merged**